### ISSUE TITLE
Bug: Using asyncio.create_subprocess_shell / _exec in lifespan raises NotImplementedError

### ISSUE THREAD

**Issue by tim-hilt** [litestar-org/litestar#3397]
### Description

Not a whole lot more to say. I want to have a long running process that updates some data periodically in the background. With FastAPI I could use the lifespan-function as presented below, but with Litestar the following exception is raised:

```
      | Traceback (most recent call last):
      |   File "C:\<disguised>\app.py", line 112, in update_data
      |     data = await exec_shell_command()
      |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |   File "C:\<disguised>\app.py", line 17, in exec_shell_command
      |     proc = await asyncio.create_subprocess_shell(
      |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |   File "C:\Appl\Python\Lib\asyncio\subprocess.py", line 208, in create_subprocess_shell
      |     transport, protocol = await loop.subprocess_shell(
      |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |   File "C:\Appl\Python\Lib\asyncio\base_events.py", line 1661, in subprocess_shell
      |     transport = await self._make_subprocess_transport(
      |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |   File "C:\Appl\Python\Lib\asyncio\base_events.py", line 502, in _make_subprocess_transport
      |     raise NotImplementedError
      | NotImplementedError
      +------------------------------------
```

I think it has something to do with the event-loop being used, but I don't know enough to find out the cause fully on my own.

### MCVE

```python
from contextlib import asynccontextmanager
from typing import AsyncGenerator

from litestar import Litestar, get
import anyio

async def update_data() -> None:
    while True:
        proc = await asyncio.create_subprocess_shell(
            "ls -l",
            stdout=asyncio.subprocess.PIPE,
            stderr=asyncio.subprocess.PIPE,
        )

        stdout, _ = await proc.communicate()
        asyncio.sleep(60)

@asynccontextmanager
async def lifespan(_: Litestar) -> AsyncGenerator[None, None]:
    async with anyio.create_task_group() as tg:
        tg.start_soon(update_data)
        yield
        tg.cancel_scope.cancel()

@get("/")
async def root() -> dict[str, str]:
    return {"Hello": "World"}

app = Litestar(route_handlers=[root], lifespan=[lifespan])
```

### Steps to reproduce

```bash
Just run the code presented above
```

### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```

### Litestar Version

2.8.2

### Platform

- [ ] Linux
- [ ] Mac
- [X] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3397">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3397/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3397/pledge.svg">
</picture>
</a>

**Comment by peterschutt** [comment]
Can you please provide the exact command that you use to run the application, e.g., `litestar run ...` or `uvicorn ...` or something else? Do you have `--reload` enabled, or are you useing multiple workers?

From [Subprocess Support on Windows](https://docs.python.org/3/library/asyncio-platforms.html#subprocess-support-on-windows):

> On Windows, the default event loop [`ProactorEventLoop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.ProactorEventLoop) supports subprocesses, whereas [`SelectorEventLoop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.SelectorEventLoop) does not.

Uvicorn explicitly set the event loop to be `SelectorEventLoop` if reloading enabled, or multiple workers ([ref](https://github.com/encode/uvicorn/blob/0efd3835da6dcc713f74aadf7b52779d0d1fa17d/uvicorn/loops/asyncio.py#L8-L10)).

> With FastAPI I could use the lifespan-function as presented below

I can find the same issue mentioned in the fastapi repo:

- https://github.com/tiangolo/fastapi/issues/4361
- https://github.com/tiangolo/fastapi/issues/964

So there has to be some difference in the configuration at the server level that is causing the `SelectorEventLoop` to be enabled in this case, and not in the FastAPI case.

**Comment by tim-hilt** [comment]
Indeed it seems to be connected to enabling multiple workers. I think I added the `workers=4` attribute to the `uvicorn.run` call when using FastAPI without even trying it out. Then I switched to Litestar and it crashed :D So: My bad. But still interesting to learn more about event-loops.

**Comment by euri10** [comment]
the tl;dr would be imho to use linux :)

now from my poor memory, uvicorn aims at supporting both windows and linux and in doing so has to deal with 2 different default event loops. both of those loop default support subprocesses, but uvicorn overrides the default one to SelectorEventLoop.
I think the override of the loop in windows is a way to let windows subprocess reload the app, with the iocp loop it wasn't working.
so in essence on windows we force the loop to be the non-default one to suport the reload thing (better dx experience) at the cost of what you encounter : being unable to use subprocesses

now this may need a rethink on uvicorn's side I'm not too sure.
